### PR TITLE
Add API support for webextension permissions

### LINKED ISSUE #3982: Expose WebExtension permissions in the API add-on version blob
We are going to start exposing WebExtension permissions on the site. We should include them in the API so that the data can be shown on the new frontend.

Since the permissions can change from version to version this should likely be exposed on the `current_version` property in add-on detail and in the full version listing.

Depends on #3981.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #3982

### CODE REVIEW

File: src/olympia/files/models.py [hunk 1]
```diff
@@ -377,21 +377,25 @@ def get_localepicker(self):
...
 
-    @amo.cached_property
+    @property
     def webext_permissions_known(self):
         return {name: perm
                 for name, perm in self.webext_permissions.iteritems()
                 if name in WEBEXT_PERMISSIONS}
 
+    @amo.cached_property(writable=True)
+    def webext_permissions_list(self):
```
Comment (diox) [inline src/olympia/files/models.py]: Could you check `self.is_webextension` before accessing `self._webext_permissions` ? That way you avoid an useless database query for non-webextension files.

It's especially interesting here, because that's what the API uses, but it might also be useful in the other properties as well.

File: src/olympia/addons/serializers.py [hunk 2]
```diff
@@ -295,19 +299,24 @@ class ESBaseAddonSerializer(BaseESSerializer):
     translated_fields = ('name', 'description', 'homepage', 'summary',
                          'support_email', 'support_url')
 
+    def fake_file_object(self, obj, data):
+        file_ = File(
+            id=data['id'], created=self.handle_date(data['created']),
+            hash=data['hash'], filename=data['filename'],
+            platform=data['platform'], size=data['size'],
+            status=data['status'], version=obj)
+        file_.webext_permissions_list = data['webext_permissions_list']
```
Comment (diox) [inline src/olympia/addons/serializers.py]: Use `data.get('webext_permissions_list', [])` in order not to break serialization with the old ES index that doesn't have that key yet.

File: src/olympia/addons/tests/test_indexers.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def test_mapping(self):
         # Make sure files mapping is set inside current_version.
         files_mapping = version_mapping['files']['properties']
         expected_file_keys = ('id', 'created', 'filename', 'hash', 'platform',
-                              'size', 'status')
+                              'size', 'status', 'webext_permissions_list')
```
Comment (diox) [inline src/olympia/addons/tests/test_indexers.py]: Add a test where it's set to something non-empty ?

File: src/olympia/addons/tests/test_serializers.py [hunk 4]
```diff
@@ -536,6 +537,19 @@ def test_no_license(self):
         assert result['id'] == self.version.pk
         assert result['license'] is None
 
+    def test_file_webext_permissions(self):
```
Comment (diox) [inline src/olympia/addons/tests/test_serializers.py]: At the moment only the tests in `AddonSerializerOutputTestMixin` are tested against both database and elasticsearch, and as a result the test you're adding here will not be tested with ES data, since it's in `TestVersionSerializerOutput`.

My suggestion (but feel free to refactor the tests differently to achieve similar or better coverage) : keep that test, but also add permissions in the basic addon created in `AddonSerializerOutputTestMixin.test_basic`, and modify `AddonSerializerOutputTestMixin._test_version` to also test permissions.

Comment (eviljeff) [inline src/olympia/addons/tests/test_serializers.py]: https://github.com/mozilla/addons-server/pull/4651/commits/103e008e18c81d7066a0f9bcb2f6928102bc74f8 (and fixes for the other comments)

**Comment by mstriemer** [comment]
API looks good.

I'm a bit curious how permissions work with hybrid add-ons. I guess there might not be a prompt since the legacy add-on would already implicitly have the permissions. Hopefully we don't need to do anything for them.

**Comment by eviljeff** [comment]
tbh I don't know anything about hybrid add-ons or what happens to them.  We might need a follow-up commit to https://github.com/mozilla/addons-server/pull/4519 that adds the parsing.

**Comment by andymckay** [comment]
I would just ignore them, they'll be gone by 57.

File: src/olympia/addons/tests/test_serializers.py (hunk 4 above)
Comment (diox) [inline src/olympia/addons/tests/test_serializers.py]: :+1:
